src/authenticator.rs

```diff
@@ -813,4 +911,48 @@ impl<T> hid::App for Authenticator<T>
         }
         Ok(())
     }
-}
\ No newline at end of file
+}
+
+
+/// Helper function to convert number sent as string to its value.
+/// ### Examples
+/// ```
+/// # use oath_authenticator::authenticator::convert_string_to_integer;
+/// assert_eq!(convert_string_to_integer(b"123"), Ok(123));
+/// assert_eq!(convert_string_to_integer(b"12345678"), Ok(12345678));
+/// ```
+/// Abort on strings longer than 8 bytes.
+/// ```
+/// use iso7816::Status;
+/// # use oath_authenticator::authenticator::convert_string_to_integer;
+/// assert_eq!(convert_string_to_integer(b"123123123123"), Err(Status::ConditionsOfUseNotSatisfied));
+/// ```
+/// Abort on invalid numbers.
+/// ```
+/// # use iso7816::Status;
+/// # use oath_authenticator::authenticator::convert_string_to_integer;
+/// assert_eq!(convert_string_to_integer(b"rubbish"), Err(Status::ConditionsOfUseNotSatisfied));
+/// assert_eq!(convert_string_to_integer(b"123h"), Err(Status::ConditionsOfUseNotSatisfied));
+/// ```
+pub fn convert_string_to_integer(str: &[u8]) -> iso7816::Result<u32> {
+    if str.len() > 8 {
+        // Handle small codes only for now
+        return Err(Status::ConditionsOfUseNotSatisfied);
+    }
+    let mut number = 0u32;
+    for e in str.iter().rev().enumerate() {
+        let (i, digit) = e;
+        if digit < &('0' as u8) || digit > &('9' as u8) {
```

Comment:
You can probably use the standard library for the test and conversion:

```rust
let Some(digit_value) = (digit as char).to_digit(10) else {
    return Err(…);
};
```

Reply:
Conversion removed entirely.